### ISSUE TITLE
Removing items with name from inventory doesn't work properly

### ISSUE THREAD

**Issue by Anoniempje1234** [SkriptLang/Skript#2714]
### Description

When you want to remove a named item from a player's inventory, it will remove the type of that item without a name as well. 

### Steps to Reproduce

1. Put this in a script:
```
command /test: 
	trigger:
		remove diamond sword named "SuperSword" from player's inventory
		message "Done!"
		stop
```
2. Get yourself a normal diamond sword. (without a name)
3. Execute the command "test".
You'll see the diamond sword is removed.
Paper 1.14.4: All diamonds swords will be removed.
Paper 1.15.1: However, if the sword has another name (Killer's Sword), it won't be removed.
### Expected Behavior

The diamond sword shouldn't be removed.
### Errors / Screenshots

/

### Server Information
* **Server version/platform:**      Paper 1.14.4 and 1.15.1
* **Skript version:**                2.4

### Additional Context

/

**Comment by Wealthyturtle** [comment]
Can you please try and see if an issue like this arises with item lore and item enchants on Skript 2.4 & Paper 1.14.4 / Skript 2.4 & Paper 1.15.1? This could be a rather large issue if ItemMeta comparisons are broken for 1.15.1 (again)

**Comment by Anoniempje1234** [comment]
Ok, I came up with this:
```
command /test [<text>]: 
	trigger:
		if arg-1 is "get":
			give player a diamond sword
			give player a diamond sword named "<reset>Useless sword"
			give player a diamond sword named "<reset>Useless sword" with lore "test"
			give player a diamond sword named "<reset>Lightningsword"
			give player a diamond sword named "<reset>Lightningsword" with lore "test"
			give player a wooden sword
			give player a wooden sword with lore "Test"
			give player a wooden sword named "test" with lore "Test"
			stop
		if arg-1 is "a":
			remove all diamond swords named "<reset>Lightningsword" from player's inventory
			message "Done!"
			stop
		if arg-1 is "b":
			remove all wood swords with lore "Test" from player's inventory
			message "Done!"
			stop
		if arg-1 is "a2":
			loop all items in player's inventory:
				loop-item is a diamond sword named "<reset>Lightningsword"
				remove loop-item from player's inventory
			stop
		if arg-1 is "b2":
			loop all items in player's inventory:
				loop-item is a wooden sword with lore "Test"
				remove loop-item from player's inventory
			stop
```

When I executed "/test a", 
the first diamond sword (the diamond sword) and the fourth diamond sword (the diamond sword named "<reset>Lightningsword"), were removed. 
When I executed "/test a2", 
only the first diamond sword (the diamond sword) was removed. 

`Only the swords with name "<reset>Lightningsword" should be removed, so only the fourth and the fifth sword.`


When I executed "/test b",
the first wooden sword (the wooden sword) and the second wooden sword (the wooden sword with lore "Test") were removed.
When I executed "/test b2", 
only the first diamond sword (the diamond sword) was removed. 

`Only the swords with lore "Test" should be removed, so only the second and the third sword.`

- I tested this on Paper 1.14.4 (Skript 2.4) and Paper 1.15.1 (Skript 2.4.1).

**Comment by Wealthyturtle** [comment]
Edited your comment for ease of reading.

**Comment by bensku** [comment]
Bug was likely introduced in 2.4-beta10 based on a report in #2727.

**Comment by madebygood** [comment]
I tried to remove an item with the name "remove all item from player's inventory", but there is a problem that cannot be removed.

**Comment by iOshawott** [comment]
hope it'll be fixed soon... :'(

**Comment by bensku** [comment]
Apologies for delay. This was top priority to me, but I had no time to work on any Skript issues for a while.

**Comment by ShaneBeee** [comment]
Fixed in 2.5-alpha2
